### PR TITLE
ML.NET rolling and fixed version CI legs

### PR DESCRIPTION
**Pull request by adamsitnik**
In #896 we have agreed that it would be good to have two different configs for ML.NET:

* one that uses latest .NET Core and fixed ML.NET to detect regressions in .NET Core
* one that uses latest version of ML.NET and LTS .NET Core to detect regressions in ML.NET

This PR tries to implement that by doing the following:

* extending the benchmarks `.yml` template with an optional `libraryVersion` parameter
* defining two jobs for ML.NET:
    * fixed ML.NET, latest .NET Core
    * latest .NET Core, fixed ML.NET
* extending the python script used by CI to set env var for controlling the version

I could not read latest ML.NET version from Maestro because ML.NET is not using Arcade, so I've used an asterisk to obtain latest preview version. 

```xml
<LibraryVersion Condition="'$(PERFLAB_LIBRARY_VERSION)' == 'rolling'">1.5.0-*</LibraryVersion>
```

This solution is simple, but requires a version bump after ML.NET increases major|minor version number (few times a year) due to https://github.com/NuGet/home/issues/912

### CODE REVIEW

**Comment by adamsitnik** [comment]
@DrewScoggins could you please take a look?

File: eng/performance/benchmark_jobs.yml [hunk 1]
```diff
@@ -92,9 +93,9 @@ jobs:
             ${{ each framework in parameters.frameworks }}:
               ${{ framework }}:
                 _Framework: ${{ framework }}
-                _Configs: CompilationMode=Tiered RunKind="${{ parameters.kind }}"
+                _Configs: CompilationMode=Tiered RunKind="${{ parameters.kind }}${{ parameters.libraryVersion }}"
```
Comment (adamsitnik) [inline eng/performance/benchmark_jobs.yml]: I don't know how the new uploader works so I've just extended the `RunKind` string with `fixed|rolling` This parameter is optional and null by default, so the "id" should change only for ML.NET runs

Comment (DrewScoggins) [reply eng/performance/benchmark_jobs.yml]: This seems like the right way to go about this.

**Comment by kunalspathak** [comment]
@DrewScoggins - after our today's discussion, should we re-evaluate this?

Also - Is it possible to update the ML.NEt benchmarks to the latest https://github.com/dotnet/machinelearning/tree/main/test/Microsoft.ML.PerformanceTests?

**Comment by danmoseley** [comment]
cc @michaelgsharp

**Comment by kunalspathak** [comment]
@adamsitnik can chime in, but I want to make sure that we remove all the [project references ](https://github.com/dotnet/machinelearning/blob/52ddbcd0a57b440e60e43e6cb39b49fe55bdfe55/test/Microsoft.ML.PerformanceTests/Microsoft.ML.PerformanceTests.csproj#L20-L27), add back nuget references and then make sure that all the benchmarks run. If you can get to that, I can verify if they run on windows/arm64 as well.

**Comment by adamsitnik** [comment]
Existing ML.NET benchmarks that we have in this [repo](https://github.com/dotnet/performance/blob/b7537def45f972d107b25e00b8200ab8837ef73a/src/benchmarks/real-world/Microsoft.ML.Benchmarks/Microsoft.ML.Benchmarks.csproj#L10) are using a hardcoded version of ML.NET in order to catch regressions in .NET itself.

The purpose of this PR was to add another CI leg that would be using latest ML.NET version to detect regressions in ML.NET itself.

@kunalspathak I assume you would be interested in updating the benchmarks, ML.NET version and using the benchmarks as part of our automated regression detection suite?

**Comment by kunalspathak** [comment]
> @kunalspathak I assume you would be interested in updating the benchmarks, ML.NET version and using the benchmarks as part of our automated regression detection suite?

That's right. It is part of our ongoing goal of having real-world benchmarks to measure the .NET performance. I also am working on adding ImageSharp to this suite in https://github.com/dotnet/performance/pull/2149. So help me understand, with the current state, if we just copy the latest code from https://github.com/dotnet/machinelearning/tree/main/test/Microsoft.ML.PerformanceTests and used nuget references to the latest ML.NET, we should be good and this PR is not blocking it.

Of course, we certainly want to track ML.NET performance and in order to do that, we need this PR.

**Comment by adamsitnik** [comment]
> we should be good and this PR is not blocking it, yes?

Yes, that is correct.

**Pull request closed without merging**